electrum_nmc/electrum/names.py

```diff
@@ -132,6 +132,296 @@ def validate_value_length(value: bytes):
+    hasher.update(b".onion checksum")
+    hasher.update(public_key)
+    hasher.update(bytes([0x03]))
+    hash_bytes = bytearray(hasher.digest())
+
+    return hash_bytes[:2]
+
+def validate_ipv4_address(address: str) -> None:
+    try:
+        ipaddress.IPv4Address(address)
+    except ipaddress.AddressValueError as e:
+        raise ValueError(f"Invalid IPv4 address: {e}")
+    except ipaddress.NetmaskValueError as e:
+        raise ValueError(f"Invalid IPv4 netmask: {e}")
+
+def validate_ipv6_address(address: str) -> None:
+    try:
+        ip = ipaddress.IPv6Address(address)
+        if address != ip.compressed:
+            raise ValueError("IPv6 address should be in compressed form")
+    except ipaddress.AddressValueError as e:
+        raise ValueError(f"Invalid IPv6 address: {e}")
+    except ipaddress.NetmaskValueError as e:
+        raise ValueError(f"Invalid IPv6 netmask: {e}")
+
+def validate_zeronet_address(address: str) -> None: # Validate P2PKH Address (Zero Net)
+    try:
+        addrtype, _ = b58_address_to_hash160(address)
+    except Exception as e:
+        raise ValueError(f"Invalid zeronet address: {e}")
+    if addrtype != ZERONET_BASE58_ADDRTYPE:
+        raise ValueError("Invalid address type.")
+
+def validate_i2p_address(address: str) -> None:
+    try:
+        # Naming rules according to https://geti2p.net/en/docs/naming
+        if address.startswith('.') or address.startswith('-'):
+            raise ValueError("Address cannot start with '.' or '-'")
+        elif '..' in address:
+            raise ValueError("Address must not contain '..'")
+        elif '.-' in address or '-.' in address:
+            raise ValueError("Sequence '.-' or '-.' is not allowed")
+        #Check for '--' except in 'xn--' for IDN
+        elif re.search(r'(?<!xn)--(?!-)', address):
+            raise ValueError("Sequence '--' is not allowed except in 'xn--'")
+        # Check if it ends with '.b32.i2p'
+        elif not address.endswith('.b32.i2p'):
+            raise ValueError("Address doesn't end with '.b32.i2p'")
+
+        # Check length of the address (According to https://geti2p.net/en/docs/naming "grep 67 characters")
+        if len(address) > 67:
+            raise ValueError("Address length exceeds 67 characters")
+        
+        # remove '.b32.i2p'
+        address = address[:-len('.b32.i2p')]
+
+        # If starts with 'xn--', remove this prefix
+        if address.startswith('xn--'):
+            address = address[4:]
+
+        # Only [a-z] [0-9] '.' and '-' are allowed
+        address_lower = address.lower()
+        pattern = r'^[a-z0-9.-]+$'
+        if not re.match(pattern, address_lower):
+            raise ValueError("Invalid characters in the address")
+
+        # Add padding
+        padding_needed = (8 - len(address) % 8)
+        padded_address = address + '=' * padding_needed
+
+        decoded = base64.b32decode(padded_address, casefold=True)
+        if len(decoded) != BASE32_LENGTH:
```

Comment:
`BASE32_LENGTH` is a really confusing constant name, since it doesn't make clear that this has something to do with I2P.